src/ebfloeseg/bbox.py

```diff
@@ -54,11 +54,25 @@ def convert(self, value, param=None, ctx=None):
             >>> BoundingBoxParser.convert("-2334051.0214676396, -414387.78951688844, -1127689.8419350237, 757861.8364224486")
             BoundingBox(x1=-2334051.0214676396, y1=-414387.78951688844, x2=-1127689.8419350237, y2=757861.8364224486)
 
+            >>> BoundingBoxParser.convert(BoundingBox(-2334051, -414387, -1127689, 757861,))
+            BoundingBox(x1=-2334051, y1=-414387, x2=-1127689, y2=757861)
 
-
+            >>> BoundingBoxParser.convert((1, 2, 3, 4))
+            Traceback (most recent call last):
+            ...
+            NotImplementedError: (1, 2, 3, 4), of type <class 'tuple'>, can't be parsed as a BoundingBox
 
 
         """
-        raw_value = ast.literal_eval(value)
-        value = BoundingBox(*raw_value)
+        if isinstance(value, BoundingBox):
+            pass  # the value is the correct type
+        elif isinstance(value, str):
+            raw_value = ast.literal_eval(value)
+            value = BoundingBox(*raw_value)
+        else:
+            msg = "%s, of type %s, can't be parsed as a BoundingBox" % (
+                value,
+                type(value),
+            )
+            raise NotImplementedError(msg)
```

Comment:
Please consider comment about this in https://github.com/WilhelmusLab/ebseg/pull/59#discussion_r1812849043

Reply:
> If you are expecting `value` to be of type `BoundingBox` or `str`, why not add a type annotation to the method signature, which will clearly communicate that intend, and just handle the conversion to that type if a string is passed?

Done. 

> ```python
def convert(self, value:BoundingBox | str, param=None, ctx=None):
```

> The string might not have the right format after all so parsing it might not succeed.

That will throw an error in the step where it tries to make the BoundingBox on line 71. That's fine.

> It feels strange to me to have an if statement with a pass in it.

I did that to avoid more than one return. I've made it into multiple returns to make the intention clearer.